addon/components/generic-rdfa-variable/insert-menu.ts

```diff
@@ -82,16 +86,32 @@ export default class GenericRdfaVariableInsertMenu extends Component<Args> {
       return;
     }
 
-    const domParser = new DOMParser();
+    this.args.controller.doCommand(this.insertCommand(editorContent), {
+      view: this.controller.mainEditorView,
+    });
+
+    this.closeModal();
+  }
+
+  insertCommand(editorContent: string): Command {
+    return (state, dispatch) => {
+      if (!this.controller || this.controller.inEmbeddedView) return false;
```

Comment:
now obsolete with my other comment, but as a rule of thumb try to avoid using the controller from inside a command function, and only use the state reference you get passed in.
It's unlikely to be a problem in practice, but there could be some funky timing issues